### PR TITLE
OCM-2741 | fix: Do not Allow IDP name cluster-admin

### PR DESCRIPTION
**Pull request by ckandag**
Ref: https://issues.redhat.com/browse/OCM-2741

### CODE REVIEW

File: cmd/create/idp/cmd.go [hunk 1]
```diff
@@ -434,6 +438,18 @@ func getIDPName(cmd *cobra.Command, idpName string, r *rosa.Runtime) string {
...
 }
 
+func validateIDPName(idpName interface{}) error {
+
+	if !idRE.MatchString(idpName.(string)) {
+		return fmt.Errorf("Invalid identifier '%s' for 'name'", idpName)
+	}
+
+	if strings.EqualFold(idpName.(string), "cluster-admin") {
+		return fmt.Errorf("The name \"cluster-admin\" is reserved for admin user IDP.")
```
[the remark this answers is not in the document]
Comment (ckandag) [reply cmd/create/idp/cmd.go]: ok

File: cmd/create/idp/cmd.go [hunk 2]
```diff
@@ -434,6 +438,18 @@ func getIDPName(cmd *cobra.Command, idpName string, r *rosa.Runtime) string {
 	return strings.Trim(idpName, " \t")
 }
 
+func validateIDPName(idpName interface{}) error {
+
+	if !idRE.MatchString(idpName.(string)) {
```
Comment (ciaranRoche) [inline cmd/create/idp/cmd.go:443]: If a non string is passed into this function it will cause a panic. 

Can we ensure `idpName` is the expected type.

**Review by ciaranRoche — COMMENTED** [review]
Just one comment inline, can we also add a test case for the `validateIDPName` function.

File: cmd/create/idp/cmd.go [hunk 3]
```diff
@@ -434,6 +438,18 @@ func getIDPName(cmd *cobra.Command, idpName string, r *rosa.Runtime) string {
 	return strings.Trim(idpName, " \t")
 }
 
+func validateIDPName(idpName interface{}) error {
```
Comment (ciaranRoche) [inline cmd/create/idp/cmd.go:441]: @oriAdler do we adopt the naming convention that CS has, which would make it `validateIdpName` or do we stick with `validateIDPName`

Comment (gdbranco) [reply cmd/create/idp/cmd.go:441]: IMO should be `Idp`

Comment (ckandag) [reply cmd/create/idp/cmd.go:441]: @ciaranRoche @gdbranco @oriAdler  -  So within the ROSACLI code, there are already several instances of using "IDP" , e.g doCreateIDP, getIDPName 
Also, I believe golang convention for acronyms is to use caps like in URL or ID etc

But if the consensus is to use Idp, I am happy to change it.

Comment (ciaranRoche) [reply cmd/create/idp/cmd.go:441]: Lets not block this MR over this, we should discuss it on the next office hours, and get consensus on what we adopt for ROSA

File: cmd/create/idp/cmd.go (hunk 2 above)
Comment (ckandag) [reply cmd/create/idp/cmd.go]: ok

**Comment by ckandag** [comment]
@ciaranRoche : Addressed the review comments, Can you pls take a look again.

File: cmd/create/idp/cmd.go [hunk 4]
```diff
@@ -434,6 +438,24 @@ func getIDPName(cmd *cobra.Command, idpName string, r *rosa.Runtime) string {
 	return strings.Trim(idpName, " \t")
 }
 
+func ValidateIdpName(idpName interface{}) error {
```
Comment (ciaranRoche) [inline cmd/create/idp/cmd.go:441]: Do we have similar validations in the API?

Comment (ckandag) [reply cmd/create/idp/cmd.go:441]: i dont think so

**Comment by ckandag** [comment]
@ciaranRoche , PTAL Again !

File: cmd/create/idp/cmd.go (hunk 4 above)
Comment (ciaranRoche) [reply cmd/create/idp/cmd.go:441]: awesome, thanks for the context. Okay makes sense to just have it here for this use case.

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**